packages/block-library/src/navigation-link/edit.js

```diff
@@ -177,7 +177,17 @@ function NavigationLinkEdit( {
 								} = {} ) => setAttributes( {
 									title: escape( newTitle ),
 									url: newURL,
-									label: label || escape( newTitle ),
+									label: ( () => {
+										const normalizedTitle = newTitle.replace( /http(s?):\/\//gi, '' );
+										const normalizedURL = newURL.replace( /http(s?):\/\//gi, '' );
+										if (
+											newTitle !== '' &&
+											normalizedTitle !== normalizedURL &&
```

Comment:
Hmmm, this might need a bit more tweaking. Now, adding a new nav item with just a url doesn't automatically fill in the label field. Or is that expected behaviour?

Reply:
It is expected because there is no label and we assume the label already there is better than replacing it with a raw link.